python/ray/services.py

```diff
@@ -582,10 +583,14 @@ def start_reaper():
     try:
         os.setpgrp()
     except OSError as e:
-        logger.warning("setpgrp failed, processes may not be "
-                       "cleaned up properly: {}.".format(e))
-        # Don't start the reaper in this case as it could result in killing
-        # other user processes.
+        if e.errno == errno.EPERM and os.getpgrp() == os.getpid():
+            # Nothing to do; we're already a session leader
+            pass
```

Comment:
Don't we also want to not `return None` in this case so the reaper gets spawned?

Reply:
Oops, yeah, thanks for the catch!